kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/VirtualKafkaClusterReconciler.java

```diff
@@ -113,8 +114,13 @@ public UpdateControl<VirtualKafkaCluster> reconcile(VirtualKafkaCluster cluster,
             VirtualKafkaCluster updatedCluster = checkClusterIngressTlsSettings(cluster, context);
             if (updatedCluster == null) {
                 updatedCluster = maybeCombineStatusWithClusterConfigMap(cluster, context);
-                Crc32ChecksumGenerator checksumGenerator = new Crc32ChecksumGenerator();
+                MetadataChecksumGenerator checksumGenerator = context.managedWorkflowAndDependentResourceContext()
+                        .get(MetadataChecksumGenerator.CHECKSUM_CONTEXT_KEY, MetadataChecksumGenerator.class)
+                        .orElse(new Crc32ChecksumGenerator());
```

Comment:
So we can inject the mock one

Reply:
I'm not spotting a unit test for this change.

Reply:
Hmm I don't have any specific tests for loading the generator but it is used by the other unit tests to verify the generator is appending what we expect